ext/intl/formatter/formatter_main.c

```diff
@@ -104,6 +110,66 @@ PHP_METHOD( NumberFormatter, __construct )
 }
 /* }}} */
 
+/* {{{ proto void NumberFormatter::__wakeup()
+ * NumberFormatter object unserializer.
+ */
+PHP_METHOD( NumberFormatter, __wakeup )
+{
+		const char* locale = intl_locale_get_default();
+		size_t      pattern_len = 0;
+		char*       pattern = NULL;
+		zend_long   style = 0;
+		UChar*      spattern     = NULL;
+		int32_t     spattern_len = 0;
+		zval        rv;
+		zval        *z_locale, *z_style, *z_pattern;
+		zend_string *zstr_locale, *zstr_pattern;
+
+		return_value = getThis();
+
+		z_locale = zend_read_property(Z_OBJCE_P(return_value), return_value, "locale", sizeof("locale") - 1, 0, &rv);
+		if (z_locale != NULL) {
+			zstr_locale = zval_get_string(z_locale);
```

Comment:
I agree with @smalyshev - it would be better to check for types and fail waking up in case of wrong data